mypy/fastparse.py

```diff
@@ -94,7 +94,7 @@ def parse_type_comment(type_comment: str, line: int) -> Type:
 
 def with_line(f: Callable[[Any, T], U]) -> Callable[[Any, T], U]:
     @wraps(f)
-    def wrapper(self, ast):
+    def wrapper(self: Any, ast: Any) -> U:
```

Comment:
Don't merge yet. I really want to have the right type for `ast` here, but I can't figure out what it is -- it looks like it should be T, which is bound by `ast35.AST`, and all the methods decorated with `@with_line` look like they take some subclass thereof, but I get the error `"T" has no attribute "lineno"` when I put that in.

I'm suspecting that this may be because there are no stubs for typed_ast, and mypy doesn't actually know what any of the classes in ast35 are?